src/main/java/org/opensearch/knn/index/codec/transfer/OffHeapByteVectorTransfer.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.transfer;
+
+import org.apache.commons.lang.StringUtils;
+import org.opensearch.knn.index.vectorvalues.KNNVectorValues;
+import org.opensearch.knn.jni.JNICommons;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Transfer byte vectors to off heap memory.
+ */
+public final class OffHeapByteVectorTransfer extends OffHeapQuantizedVectorTransfer<byte[], byte[]> {
+
+    // TODO: Replace with KNNByteVectorValues
+    public OffHeapByteVectorTransfer(KNNVectorValues<byte[]> vectorValues, final Long batchSize) throws IOException {
+        super(vectorValues, batchSize, (vector, state) -> vector, StringUtils.EMPTY, DEFAULT_COMPRESSION_FACTOR);
+    }
+
+    // TODO: Replace with KNNByteVectorValues
+    public OffHeapByteVectorTransfer(KNNVectorValues<byte[]> vectorValues) throws IOException {
+        this(vectorValues, null);
```

Comment:
what will be the case when we need to pass the batchSize, just wondering?

Reply:
Yes. [VectorTransferIndexBuildStrategy](https://github.com/opensearch-project/k-NN/pull/1950/files#diff-6047a6c31b628a0edc4bd7eba164bdfc35fcc53de703a25a295e91aa04b94d20R85)

Reply:
thanks found it later. Its hard to grasp from code.